File: Solutions/167. Two Sum II - Input Array Is Sorted/167. Two Sum II - Input Array Is Sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int lo=0,hi=nums.size()-1;
        while(lo<hi){
            int tmp=nums[lo]+nums[hi];
            if(tmp==target){
                return {lo+1,hi+1};
            }
            if(tmp<target){
                ++lo;
            }else{
                --hi;
            }
        }
        return {-1,-1};
    }
};

File: Solutions/167. Two Sum II - Input Array Is Sorted/167. Two Sum II - Input Array Is Sorted.go
func twoSum(nums []int, target int) []int {
    lo,hi:=0,len(nums)-1
    for lo<hi{
        tmp:=nums[lo]+nums[hi]
        if tmp==target{
            return []int{lo+1,hi+1}
        }
        if tmp<target{
            lo+=1
        }else{
            hi-=1
        }
    }
    return []int{-1,-1}
}